src/main/java/uk/ac/ebi/eva/pipeline/io/writers/VepAnnotationFileWriter.java

```diff
@@ -0,0 +1,225 @@
+            flushToPerlStdin();
+            waitUntilProcessEnds(timeoutInSeconds);
+            checkExitStatus();
+            checkOutputWritingStatus();
+            logger.info("VEP process finished");
+            alreadyClosed = true;
+        }
+    }
+
+    private void flushToPerlStdin() {
+        try {
+            perlStdin.flush();
+            perlStdin.close();
+        } catch (IOException e) {
+            logger.error("Could not close stream for VEP's stdin", e);
+        }
+    }
+
+    private void waitUntilProcessEnds(Long timeoutInSeconds) {
+        boolean finished;
+        try {
+            finished = process.waitFor(timeoutInSeconds, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            throw new ItemStreamException(e);
+        }
+
+        if (!finished) {
+            String timeoutReachedMessage = "Reached the timeout (" + timeoutInSeconds
+                    + " seconds) while waiting for VEP to finish. Killed the process.";
+            logger.error(timeoutReachedMessage);
+            process.destroy();
+            throw new ItemStreamException(timeoutReachedMessage);
+        }
+    }
+
+    private void checkExitStatus() {
+        int exitValue = process.exitValue();
+        if (exitValue != 0) {
+            String errorLog = annotationParameters.getVepOutput() + ".errors.txt";
+            try {
+                connectStreams(new BufferedInputStream(process.getErrorStream()), new FileOutputStream(errorLog));
+            } catch (IOException e) {
+                throw new ItemStreamException("VEP exited with code " + exitValue
+                        + " but the file to dump the errors could not be created: " + errorLog,
+                        e);
+            }
+            throw new ItemStreamException("Error while running VEP (exit status " + exitValue + "). See "
+                    + errorLog + " for the errors description from VEP.");
+        }
+    }
+
+    private void checkOutputWritingStatus() {
+        try {
+            final long CONVERT_SECONDS_TO_MILLISECONDS = 1000L;
+            outputCapturer.join(timeoutInSeconds * CONVERT_SECONDS_TO_MILLISECONDS);
+        } catch (InterruptedException e) {
+            throw new ItemStreamException("Interrupted while waiting for the VEP output writer thread to finish. ", e);
+        }
+        if (outputCapturer.isAlive()) {
+            outputCapturer.interrupt();
+            throw new ItemStreamException("Reached the timeout (" + timeoutInSeconds
+                    + " seconds) while waiting for VEP output writing to finish. Killed the thread.");
+        }
+        if (!writingOk.get()) {
+            throw new ItemStreamException("VEP output writer thread could not finish properly. ");
+        }
+    }
+
+    /**
+     * read all the inputStream and write it into the outputStream
+     */
+    private long connectStreams(InputStream inputStream, OutputStream outputStream) throws IOException {
```

Comment:
I would give a second thought to this method. It's not only that the return value is not used, also, the written only conts the number of successful reads. This will not be equal to the full readed size in some scenarios. Not only that, this method is doing a hungry active waiting. It should sleep from time to time or you are just gonna have a lot of wasted cycles.